typescript/agentkit/src/wallet-providers/privyWalletProvider.ts

```diff
@@ -0,0 +1,131 @@
+import { PrivyClient } from "@privy-io/server-auth";
+import { createViemAccount } from "@privy-io/server-auth/viem";
+import { ViemWalletProvider } from "./viemWalletProvider";
+import { createWalletClient, http, WalletClient } from "viem";
+import { NETWORK_ID_TO_VIEM_CHAIN } from "../network/network";
+
+/**
+ * Configuration options for the Privy wallet provider.
+ *
+ * @interface
+ */
+interface PrivyWalletConfig {
+  /** The Privy application ID */
+  appId: string;
+  /** The Privy application secret */
+  appSecret: string;
+  /** The ID of the wallet to use */
+  walletId?: string;
+  /** Optional network ID to connect to */
+  networkId?: string;
```

Comment:
@0xRAG @azf20 - This config should take `chainId` instead of `networkId` . Network ID is a CDP concept and will limit the Privy implementation to only support CDP supported networks. Privy does not have the same constraints as `CdpWalletProvider` and therefore should take `chainId` which will unblock usage of all EVM.

Reply:
Good call! This is done now in https://github.com/coinbase/agentkit/pull/242/commits/ba189e3d19a4cf98c31c86e52db2dacd35daae54